### PR TITLE
Parse CSV

### PR DESCRIPTION
**Pull request by Ammar022**
- Added functionality to read and parse .csv files given as command line argument. 
- Added functionality to print data in JSON Format.
- Updated README.md.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+func main() {
+	// Open the .csv file
+	file, err := os.Open(os.Args[1])
```
Comment (mzeeshanali) [inline main.go:12]: what if user does not pass filepath?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	// Read all records from the .csv file
+	records, err := reader.ReadAll()
+	if err != nil {
+		fmt.Println("Failed to read records:", err)
+		os.Exit(1)
+	}
+
+	//Converting data in json format
+	s, _ := json.Marshal(records)
```
Comment (mzeeshanali) [inline main.go:30]: handle error please

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (Ammar022) [reply main.go:12]: It will return error I have checked this condition.

**Review by mzeeshanali — CHANGES_REQUESTED** [review]
Update your `PR` title please

File: main.go (hunk 2 above)
Comment (Ammar022) [reply main.go:30]: Done.

File: main.go (hunk 1 above)
Comment (mzeeshanali) [reply main.go:12]: what error it will return? what if args length is less than two?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	"fmt"
+	"os"
+)
+
+func main() {
+	// Check if a file path was passed as a command-line argument
+	if len(os.Args) != 2 {
+		fmt.Println("Please provide a file path")
+		os.Exit(1)
+	}
```
Comment (mzeeshanali) [inline main.go:15]: what if in future we want to get one additional argument along with filepath?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	records, err := reader.ReadAll()
+	if err != nil {
+		fmt.Println("Failed to read records:", err)
+		os.Exit(1)
+	}
+
+	//Converting data in json format
+	s, err := json.Marshal(records)
+	if err != nil {
+		fmt.Println("Failed to convert Data in Jason:", err)
```
Comment (mzeeshanali) [inline main.go:38]:
```suggestion
		fmt.Println("Failed to marshal data into json:", err)
```

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

File: data2.csv [hunk 5]
```diff
@@ -0,0 +1 @@
+heelo, world,24
```
Comment (mzeeshanali) [inline data2.csv:1]: add empty line

File: main.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		records, err := reader.ReadAll()
+		if err != nil {
+			fmt.Println("Failed to read the CSV file:", err)
+			os.Exit(1)
+		}
+
+		// Converting data in json format
+		s, err := json.Marshal(records)
+		if err != nil {
+			fmt.Println("Failed to marshal Data in Jason:", err)
```
Comment (mzeeshanali) [inline main.go:43]:
```suggestion
			fmt.Println("Failed to marshal data in json:", err)
```

File: main.go [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+			os.Exit(1)
```
Comment (mzeeshanali) [inline main.go:44]: why `os.Exit(1)` here?

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

**Review by mzeeshanali — APPROVED** [review]

**Pull request merged**